scripts/automation/coverage/run.py

```diff
@@ -0,0 +1,122 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import os.path
+import sys
+import itertools
+
+from coverage import Coverage
+
+import azure.cli.core.application as cli_application
+from ..tests.nose_helper import get_nose_runner
+from ..utilities.path import get_core_modules_paths_with_tests, \
+    get_command_modules_paths_with_tests, get_repo_root, get_test_results_dir, make_dirs
+
+
+# TODO: Fix track command logic in vcr_test_base.py.
```

Comment:
I'm assuming this is the traditional line or branch based code coverage metric, or is it a port of the command coverage logic (which needs to be augmented, but that's a separate thing).